app/services/onlinetesting/phase1/Phase1TestEvaluation.scala

```diff
@@ -26,34 +27,43 @@ import services.onlinetesting.OnlineTestResultsCalculator
   }
 
-  private def evaluate(isGis: Boolean, schemes: List[SchemeId], passmark: Phase1PassMarkSettings,
-               sjqTestResult: TestResult, bqTestResultOpt: Option[TestResult] = None) = {
-    for {
+  private def evaluate(applicationRoute: ApplicationRoute, isGis: Boolean, schemes: List[SchemeId], passmark: Phase1PassMarkSettings,
+                       sjqTestResult: TestResult, bqTestResultOpt: Option[TestResult] = None): List[SchemeEvaluationResult] = {
+    val evaluationResults = for {
       schemeToEvaluate <- schemes
-      schemePassmark <- passmark.schemes find (_.schemeId == schemeToEvaluate)
     } yield {
-      val sjqResult = evaluateTestResult(schemePassmark.schemeThresholds.situational)(sjqTestResult.tScore)
-      val bqResult = bqTestResultOpt.map(_.tScore).map(evaluateTestResult(schemePassmark.schemeThresholds.behavioural)).getOrElse(Green)
-      Logger.debug(s"Processing scheme $schemeToEvaluate, " +
-        s"sjq score = ${sjqTestResult.tScore}, " +
-        s"sjq fail = ${schemePassmark.schemeThresholds.situational.failThreshold}, " +
-        s"sqj pass = ${schemePassmark.schemeThresholds.situational.passThreshold}, " +
-        s"sqj result = $sjqResult, " +
-        s"bq score = ${bqTestResultOpt.map(_.tScore).getOrElse(None)}, " +
-        s"bq fail = ${schemePassmark.schemeThresholds.behavioural.failThreshold}, " +
-        s"bq pass = ${schemePassmark.schemeThresholds.behavioural.passThreshold}, " +
-        s"bq result = $bqResult"
-      )
-      SchemeEvaluationResult(schemeToEvaluate, combineTestResults(sjqResult, bqResult).toString)
+      val schemePassmarkOpt = passmark.schemes.find(_.schemeId == schemeToEvaluate)
+      schemePassmarkOpt.map { schemePassmark =>
+        val sjqResult = evaluateTestResult(schemePassmark.schemeThresholds.situational)(sjqTestResult.tScore)
+        val bqResult = bqTestResultOpt.map(_.tScore).map(evaluateTestResult(schemePassmark.schemeThresholds.behavioural)).getOrElse(Green)
+        Logger.debug(s"Processing scheme $schemeToEvaluate, " +
+          s"sjq score = ${sjqTestResult.tScore}, " +
+          s"sjq fail = ${schemePassmark.schemeThresholds.situational.failThreshold}, " +
+          s"sqj pass = ${schemePassmark.schemeThresholds.situational.passThreshold}, " +
+          s"sqj result = $sjqResult, " +
+          s"bq score = ${bqTestResultOpt.map(_.tScore).getOrElse(None)}, " +
+          s"bq fail = ${schemePassmark.schemeThresholds.behavioural.failThreshold}, " +
+          s"bq pass = ${schemePassmark.schemeThresholds.behavioural.passThreshold}, " +
+          s"bq result = $bqResult"
+        )
+        Option(SchemeEvaluationResult(schemeToEvaluate, combineTestResults(sjqResult, bqResult).toString))
+      }.getOrElse {
+        if (Scheme.isSdip(schemeToEvaluate) && applicationRoute == ApplicationRoute.SdipFaststream) {
+          Option(SchemeEvaluationResult(schemeToEvaluate, Amber.toString))
```

Comment:
I'd make this a Some(), i believe Options are only best used when you're dealing with something that might be null